app/code/Magento/Catalog/Model/Indexer/Product/Flat/Action/Row.php

```diff
@@ -41,7 +48,8 @@ public function __construct(
         TableBuilder $tableBuilder,
         FlatTableBuilder $flatTableBuilder,
         Indexer $flatItemWriter,
-        Eraser $flatItemEraser
+        Eraser $flatItemEraser,
+        MetadataPool $metadataPool
```

Comment:
@lewisvoncken, due to [Magento backward-compatible guide](http://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/) we can't add a new required dependency to the constructor.

Reply:
@VladimirZaets  thanks for the feedback. The dependency incorrect and I have just removed it.